crates/consensus/beacon/src/engine/mod.rs

```diff
@@ -586,9 +586,10 @@ where
                     debug!(target: "consensus::engine", fcu_head_num=?outcome.header().number, current_head_num=?self.blockchain.canonical_tip().number, "Ignoring beacon update to old head");
                 }
 
-                if let Some(attrs) = attrs {
+                if let Some((builder, attrs)) = self.payload_builder.as_ref().zip(attrs) {
```

Comment:
is this okay with engine api?
that payload building is optional?

Reply:
The payload building is not optional, hence this behavior is not the default one.

However, the ability to disable _default_ payload building is beneficial for custom integrations.

Reply:
> However, the ability to disable default payload building is beneficial for custom integrations.

what would be a custom integration?